osu.Game/Overlays/RestoreDefaultValueButton.cs

```diff
@@ -20,15 +20,30 @@ public class RestoreDefaultValueButton<T> : OsuButton, IHasTooltip, IHasCurrentV
     {
         public override bool IsPresent => base.IsPresent || Scheduler.HasPendingTasks;
 
-        private readonly BindableWithCurrent<T> current = new BindableWithCurrent<T>();
-
         // this is done to ensure a click on this button doesn't trigger focus on a parent element which contains the button.
         public override bool AcceptsFocus => true;
 
+        // this is intentionally not using BindableWithCurrent as this needs a BindableNumber instance for an accurate IsDefault value.
+        //
+        // this also cannot use IBindableWithCurrent.Create() due to BindableNumberWithCurrent
+        // directly casting given bindables to BindableNumber, which is not necessarily the case.
+        //
+        // therefore rely on the old method of taking each current bindable instance for now, until things are settled framework-side.
```

Comment:
I can't grasp this comment. How about

```suggestion
        // this is intentionally not using BindableWithCurrent, as it can use the wrong IsDefault implementation when passed a BindableNumber.
        // using GetBoundCopy() ensures that the received bindable is of the exact same type as the source bindable and uses the proper IsDefault implementation.
```

Reply:
This doesn't seem to mention why `IBindableWithCurrent.Create()` isn't used but I guess it's a small unwanted detail, seems more simplified.